src/version.rs

```diff
@@ -259,4 +294,71 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_use_package_version_as_hash_key() -> Result<()> {
```

Comment:
Is this just not just putting strings into an hashmap?

Reply:
Yes, that's why it's a draft

Reply:
Ah

Reply:
done